### PR TITLE
examples: update @clerk/tanstack-start to 0.3.5

### PR DESCRIPTION
**Pull request by zwgnr**
- updates @clerk/tanstack-start to latest version `0.3.5`

### CODE REVIEW

File: examples/react/start-clerk-basic/app/routes/_authed.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export const Route = createFileRoute('/_authed')({
     if (error.message === 'Not authenticated') {
       return (
         <div className="flex items-center justify-center p-12">
-          <SignIn forceRedirectUrl={window.location.href} />
+          <SignIn routing='hash' forceRedirectUrl={window.location.href} />
```
Comment (SeanCassiere) [inline examples/react/start-clerk-basic/app/routes/_authed.tsx:14]: Any reason why `routing="hash"` is required here?

Comment (zwgnr) [reply examples/react/start-clerk-basic/app/routes/_authed.tsx:14]: Not entirely sure to be honest. After updating clerk version, when _not_ signed in and trying to navigate from /posts -> home the routing is broken and I was getting "not found". Adding routing="hash" fixed the issue for me

Comment (SeanCassiere) [reply examples/react/start-clerk-basic/app/routes/_authed.tsx:14]: Hmmm, alright. Could you fix the merge conflicts? I'll try it out before merging.

Comment (zwgnr) [reply examples/react/start-clerk-basic/app/routes/_authed.tsx]: fixed merge conflicts. if you aren't encountering the issue without the "hash" change, or have a better solution please let me know!

**Pull request merged**